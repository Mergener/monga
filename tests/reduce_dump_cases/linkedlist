type Node = {
	value : int;
	next : Node;
};

function newNode(value : int): Node {
	var ret : Node;

	ret = new Node;

	ret.value = value;
	ret.next = null;

	return ret;
}