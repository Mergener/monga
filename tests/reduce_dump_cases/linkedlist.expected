type r1
def_field r1
field_defs r1
type r1
def_field r1
field_defs r2
typedesc r3
def_type r1
definition r3
definitions r1
type r1
parameter r1
parameters r1
opt_parameters r1
type r1
opt_ret_type r2
type r1
def_variable r1
variable_defs r1
opt_variable_defs r2
var r1
exp_primary r3
exp_member r1
type r1
opt_bracket_exp r1
exp_conditional r3
exp r1
block r3
statements r1
var r1
exp_primary r3
exp_member r1
var r1
exp_primary r3
exp_member r2
var r1
exp_primary r3
exp_member r1
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
block r3
statements r2
var r1
exp_primary r3
exp_member r1
var r1
exp_primary r3
exp_member r2
var r1
exp_primary r3
exp_member r1
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
block r3
statements r2
var r1
exp_primary r3
exp_member r1
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
opt_exp r2
block r4
statements r2
opt_statements r2
block r1
def_function r1
definition r2
definitions r2
module r1